src/Twig/Extension/AuthRuntime.php

```diff
@@ -191,6 +191,54 @@ public function getProviders($guid = null)
     }
 
     /**
+     * Return an array of registered OAuth providers for an account with
+     * lst steen and last updated dates
+     *
+     * @param string $guid
+     *
+     * @return array
+     */
+    public function getProvidersDates($guid = null)
+    {
+        $providers = [];
+        if ($guid === null) {
+            $auth = $this->session->getAuthorisation();
+
+            if ($auth === null) {
+              return $providers;
+            }
+            $guid = $auth->getGuid();
+        }
+
+        $providerEntities = $this->records->getProvisionsByGuid($guid);
+        if ($providerEntities === false) {
+            return $providers;
+        }
+
+        /** @var Storage\Entity\Provider $providerEntity */
+        $providers['lastseen'] = null;
+        $providers['lastupdate'] = null;
```

Comment:
Yes.

supposing you have a few providers you would get a datastructure that is something like:
```php
$providers = [
    0 => [ 
        'name' => 'google',
        'lastseen' => 'some date',
        'lastupdate' => 'some date'
    ],
    1 => [ 
        'name' => 'github',
        'lastseen' => 'some later date',
        'lastupdate' => 'some later date'
    ],
    'lastseen' => 'some date',
    'lastupdate' => 'some date'
];
```
Ususally you're only interested in the smallest last seen date, but it made no sense to me to throw away the other data.

The rest of the keys is structured similar to `getProviders()` .. which also returns a simple array of providers.